packages/sanity/src/core/form/useDocumentForm.ts

```diff
@@ -259,12 +267,21 @@ export function useDocumentForm(options: DocumentFormOptions): DocumentFormValue
   )
 
   const requiredPermission = value._createdAt ? 'update' : 'create'
+  const getDocumentId = useCallback(
+    (docId: string) => {
+      if (liveEdit || isPublishedId(docId)) {
```

Comment:
I find it somewhat hard to reason about the passed in docId vs the returned documentId. What constraints are applied to the one we get passed? E.g. can it ever be a version id?